UKHomeOffice/nmsw-ui
# NMSW-1059 Test cases for more actions form & related

# Ticket



----

## AC
**AC - Exit without doing a next action & return via task list page**
GIVEN I have uploaded a FAL 6
WHEN I am still awaiting more passenger details to check
THEN I am able to exit the page without further action

WHEN I return to the task list page
THEN the task status shows 'In progress'

**AC - Select to upload FAL 6 again**
GIVEN I am viewing a draft voyage with an uploaded FAL 6
WHEN I select to upload a FAL6
THEN the last FAL6 upload and UPT statuses are displayed
AND I am able to upload another FAL6

WHEN I have uploaded a new FAL6
THEN the status page shows the new filename
THEN the status page should the new count of passengers
THEN the status page shows the list of passengers from the new file

**AC - Upload passenger details**

GIVEN I have uploaded and checked a FAL6 
WHEN I have NOT selected this to be the final file
WHEN I 'save and comeback later'
THEN I return to the task list page
AND the task status shows 'In progress'

**AC - Finish checking passengers**

GIVEN I have uploaded and checked a FAL6 
WHEN I 'Finish checking passengers'
THEN I return to the task list page
AND the task status has changed to 'Complete'

GIVEN I have a declaration where I have chosen final file
WHEN I view the task list for that declaration
THEN it shows passengers as completed

**AC - Save and come back later should not change the passenger status**

GIVEN I have uploaded and checked a FAL6
WHEN I click save and come back later
THEN the task list page should show 'in progress'

GIVEN I select final for that file
WHEN I click save and continue
THEN the task list page should show complete
WHEN I click passengers link to return to the UPT Status check page
WHEN I click save and come back later
THEN the task list page should still show complete

----

## To test

- create a new declaration and upload a FAL6
- view the UPT status results page
- leave the page
- go back to the voyage and see the task list page
- > the passenger status should be 'In progress'
- click the passenger link to return to the UPT status checking page
- > there should be a link to `save and come back later`
- click the link
- > you should be taken to the Task List page
- > the passenger status should be 'In progress'
- click on passengers to return to the UPT status results page
- > there should be a more actions form with an option to upload a new fal6
- select the option to upload a new fal 6
- click save and continue
- > you should be taken to the upload passengers page
- select yes
- upload a new fal 6
- continue on to the UPT status results page
- > you should see the new filename
- > you should see the new count
- > you should see a list of passengers that matches your latest file
- select Final in the more action radio list
- click save and continue
- > you should be taken to the task list page
- > the passenger status should be complete
- leave the page
- return to that declarations Task list page
- > the passenger status should be complete
- click on passengers to go to the UPT status results page
- click on save and come back later
- > you should be taken to the task list page
- > the passenger status should be complete

----

## Notes
